libs/vkd3d/command.c

```diff
@@ -1779,11 +1779,13 @@ static void d3d12_command_list_end_current_render_pass(struct d3d12_command_list
     }
 
     if (list->current_render_pass)
-    {
         VK_CALL(vkCmdEndRenderPass(list->vk_command_buffer));
+
+    /* Don't emit barriers for temporary suspendion of the render pass */
+    if (!suspend && (list->current_render_pass || list->render_pass_suspended))
         d3d12_command_list_emit_render_pass_transition(list, VKD3D_RENDER_PASS_TRANSITION_MODE_END);
-    }
 
+    list->render_pass_suspended = suspend && list->current_render_pass;
```

Comment:
Review not to self: Do we perform suspended barriers on vkEndCommandBuffer?

Reply:
We should, since `Close` calls `d3d12_command_list_end_current_render_pass(..., false)`.

Reply:
Yes